### PR TITLE
Support to evaluate on PAI

### LINKED ISSUE #2148: Support run TO EVALUATE on PAI
As mentioned https://github.com/sql-machine-learning/sqlflow/pull/2139, need to support run TO EVALUATE on PAI.

### PR DESCRIPTION
**Pull request by typhoonzero**
Fix https://github.com/sql-machine-learning/sqlflow/issues/2148

### CODE REVIEW

File: python/sqlflow_submitter/tensorflow/evaluate.py [hunk 1]
```diff
@@ -76,7 +76,7 @@ def evaluate(datasource,
     model_params.update(feature_columns)
     if is_estimator:
         if is_pai:
-            define_tf_flags()
+            # define_tf_flags()
```
Comment (shendiaomo) [inline python/sqlflow_submitter/tensorflow/evaluate.py]: Remove this line or add a TODO?

Comment (typhoonzero) [reply python/sqlflow_submitter/tensorflow/evaluate.py]: Deleted this line, flags will be defined in the PAI code generator.

File: pkg/sql/codegen/pai/tensorflow.go [hunk 2]
```diff
@@ -112,6 +112,39 @@ func TFLoadAndExplain(ir *ir.ExplainStmt, session *pb.Session, modelPath string,
...
+func TFLoadAndEvaluate(ir *ir.EvaluateStmt, session *pb.Session, modelPath string, expn *ExplainRender) (string, error) {
+	var tpl = template.Must(template.New("Evaluate").Parse(tfEvaluateTmplText))
+	ossModelDir := OSSModelURL(modelPath)
+	paiExplainTable := ""
+	if tensorflow.IsPAI() && ir.TmpEvaluateTable != "" {
+		paiExplainTable = ir.TmpEvaluateTable
+	}
+
+	validationMetrics := ""
+	for k, v := range ir.Attributes {
```
Comment (shendiaomo) [inline pkg/sql/codegen/pai/tensorflow.go]: Can we  use an `if` statement here instead? Like
```go
if v, ok := ir.Attributes["validationMetrics"]; ok && len(ir.Attributes) == 1 {
    validationMetrics = v.(string)
} else if len(ir.Attributes) != 0 {
    // Output diagnostics messages
}
```

File: pkg/sql/codegen/pai/tensorflow.go [hunk 3]
```diff
@@ -112,6 +112,39 @@ func TFLoadAndExplain(ir *ir.ExplainStmt, session *pb.Session, modelPath string,
...
+	}
+
+	filler := evaluateFiller{
+		OSSModelDir:       ossModelDir,
+		DataSource:        session.DbConnStr,
+		Select:            ir.Select,
+		ResultTable:       ir.Into,
+		IsPAI:             tensorflow.IsPAI(),
+		PAITable:          paiExplainTable,
+		ValidationMetrics: validationMetrics,
```
Comment (shendiaomo) [inline pkg/sql/codegen/pai/tensorflow.go:138]: There're so many `Filler`s like this, later we have to find a better way to organize these.

File: pkg/sql/codegen/pai/tensorflow.go (hunk 2 above)
Comment (typhoonzero) [reply pkg/sql/codegen/pai/tensorflow.go]: Done.

File: pkg/sql/pai_submitter.go [hunk 4]
```diff
@@ -310,6 +310,72 @@ func (s *paiSubmitter) ExecuteExplain(cl *ir.ExplainStmt) error {
...
+		return err
+	}
+	expn, e := pai.Evaluate(cl, s.Session, scriptPath, paramsPath, cl.ModelName, ossModelPath, s.Cwd, modelType)
+	if e != nil {
+		return e
+	}
+	if e = s.submitPAITask(expn.Code, expn.PaiCmd, expn.Requirements); e != nil {
+		return e
+	}
+	if img, e := expn.Draw(); e == nil {
```
Comment (Yancey0623) [inline pkg/sql/pai_submitter.go]: Why `Evaluate` outputs a figure?

**Review by Yancey0623 — APPROVED** [review]
LGTM

**Pull request merged**